### ISSUE TITLE
How to change default file's extension that will be saved?

### ISSUE THREAD

**Issue by liwenjing93** [EslaMx7/PasteIntoFile#15]
Hi,  I regularly save images from clipboard, and I like to save them with .Jpg instead of .Png format, because .jpg will have a small file size than .png. However, everytime I want to save an image, I have to choose AGAIN .jpg because the default option is always .png. It has always annoyed me a lot. 

![image](https://user-images.githubusercontent.com/9654732/74172845-88464000-4c63-11ea-8fc2-2de3f256ff93.png)

I hope you could make some changes into that software in the next release, so that I could change the default option from png to jpg or another extension.

Thank you so much, your software is really useful. I love it so much. 

Best Regards,

**Comment by EslaMx7** [comment]
@liwenjing93 Thank you very much for your feedback!
I will make sure to have this issue fixed in the next release very soon.

**Comment by liwenjing93** [comment]
> I will make sure to have this issue fixed in the next release very soon.
Thank you!!!

**Comment by edwardjohnson11** [comment]
Choosing a default would be great!

Additionally, would also be great if an underscore is used in the automatically generated file name, between the date and the time, so as to avoid hyperlink issues: 13-04-2020_21-38-23

**Comment by subz390** [comment]
BUMP +1 on being able to set the default file extension. 
For the majority of saving applications I choose `jpg` as it takes up less space.

**Comment by barn852** [comment]
Yes, it would be so good to change or have option for default image format. I think most people use jpg for any kind of image, except for some specific purpose.

**Comment by subz390** [comment]
For anyone who's interested here's an [AutoHotkey](https://www.autohotkey.com/) script that'll send keyboard inputs to select the `jpg` save as extension and press `enter`.
1. Install AutoHotkey
2. Copy the script into a file with an `.ahk` extension and double click the file to run it.
3. To start up with Windows put the file in the Windows `startup` folder.

For reference I've added the modifier keys in comments so you can set your key combination preference.


```ahk
#SingleInstance, Force
SendMode Input ; Forces Send and SendRaw to use SendInput buffering for speed.
SetWorkingDir, %A_ScriptDir%
#MaxThreadsPerHotkey, 1 ; no re-entrant hotkey handling

; ! Alt
; + Shift
; ^ Ctrl
; ^+ Ctrl Shift

; ====================================================
; PasteIntoFile
; sets the save as extension to jpg
; Ctrl Shift j
#IfWinActive ahk_exe PasteIntoFile.exe
^+j::
    Send, {Tab}{Down}{Tab}{Tab}{Tab}{Tab}{Enter}
return
```
![ahkinstructions](https://user-images.githubusercontent.com/11546815/150636758-6879c0ad-eec3-4730-8e9f-3bf1336093a8.jpg)

**Comment by barn852** [comment]
Thanks @subz390 for the idea of using AutoHotkey. I have modified it so the script changes png to jpg any time PasteIntoFile is executed. Maybe it is not the nicest solution, but it works. If you want to use it for other extension, just replace png/jpg accordingly.

```
#SingleInstance, Force
While 1{
  WinWaitActive ahk_exe PasteIntoFile.exe
  ControlGetText Val,Edit1,ahk_exe PasteIntoFile.exe
  If (Val="png")
    ControlSend Edit1,{Down},ahk_exe PasteIntoFile.exe
}
```
update1: changed the code, the previous did not work well
update2: I have just found [eltos' fork](https://github.com/eltos/PasteIntoFile) that has a config to set the default image format. No need for the the ahk script.
